collector/lib/Logging.h

```diff
@@ -71,25 +77,79 @@ class LogMessage {
               << std::left << std::setw(LevelPaddingWidth) << GetLogLevelName(level_)
               << " " << std::put_time(nowTm, "%Y/%m/%d %H:%M:%S")
               << "] ";
+  }
 
-    if (throttled_) {
-      std::cerr << "[Throttled] ";
-    }
-
-    if (include_file_) {
+  void PrintFile() {
+    if (CheckLogLevel(LogLevel::DEBUG)) {
       const char* basename = strrchr(file_, '/');
```

Comment:
It's strictly speaking beyond this PR, but since we have c++17, why don't we use something fancy here, like `std::filesystem::path` instead of a plain string?

Reply:
I didn't want to do too many changes here outside of counting logs, but I agree that using `std::filesystem::path` makes more sense here, I'll take a look into it.